Panic error after upgrading from 1.7.2 to 1.8.3

Greetings,

We have a livekit setup (runs as docker-compose container)  which worked fine with verison 1.7.2 but after upgrading the container from version 1.7.2 to 1.8.3, the container repeatedly restarts due to a runtime error.  Each time the server starts, it initializes (e.g., Redis connection, TURN server startup) but crashes with a `panic: runtime error: invalid memory address or nil pointer dereference`.

Logs:

```
2025-01-28T04:14:43.025Z        INFO    livekit service/server.go:258   starting LiveKit server {"portHttp": 7880, "nodeID": "ND_uCEm2vn6tDHd", "nodeIP": "212.46.61.44", "version": "1.8.3", "rtc.portTCP": 7881, "rtc.portICERange": [50000, 60000], "portPrometheus": 6789}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x12c65b8]

goroutine 249 [running]:
github.com/livekit/livekit-server/pkg/service.(*defaultSessionHandler).HandleSession(0xc001044c90, {0x1909038, 0xc0017199e0}, {{0xc00159fd90, 0x9}, {0x0, 0x0}, 0x0, 0x0, 0x1, ...}, ...)
        /workspace/pkg/service/signal.go:101 +0x78
github.com/livekit/livekit-server/pkg/service.(*signalService).RelaySignal(0xc000cff3b0, {0x7cf7812e94d8, 0xc0011059e0})
        /workspace/pkg/service/signal.go:180 +0x671
github.com/livekit/psrpc/pkg/server.(*streamHandler[...]).handleOpenRequest(0x1920800, 0xc00105e300, 0xc000636070, 0xc0003fab00)
        /go/pkg/mod/github.com/livekit/psrpc@v0.6.1-0.20241018124827-1efff3d113a8/pkg/server/stream.go:202 +0x4ca
github.com/livekit/psrpc/pkg/server.(*streamHandler[...]).handleRequest.func1()
        /go/pkg/mod/github.com/livekit/psrpc@v0.6.1-0.20241018124827-1efff3d113a8/pkg/server/stream.go:141 +0x34
created by github.com/livekit/psrpc/pkg/server.(*streamHandler[...]).handleRequest in goroutine 227
        /go/pkg/mod/github.com/livekit/psrpc@v0.6.1-0.20241018124827-1efff3d113a8/pkg/server/stream.go:140 +0xd3
2025-01-28T04:14:52.284Z        INFO    livekit redis/redis.go:142      connecting to redis     {"simple": true, "addr": "10.5.86.1:6379"}
2025-01-28T04:14:53.431Z        INFO    livekit rtcconfig/webrtc_config.go:262  failed to get external ip       {"local": "172.17.0.1", "err": "context canceled"}
2025-01-28T04:14:53.431Z        INFO    livekit rtcconfig/webrtc_config.go:262  failed to get external ip       {"local": "10.5.86.4", "err": "context canceled"}
2025-01-28T04:14:53.431Z        INFO    livekit rtcconfig/webrtc_config.go:88   using external IPs      {"ips": ["212.46.61.44/212.46.61.44", "10.5.86.4/10.5.86.4", "172.17.0.1/172.17.0.1"]}
2025-01-28T04:14:53.432Z        INFO    livekit.pion.tcp_mux    v4@v4.0.3/tcp_mux.go:99 Listening TCP on [::]:7881
2025-01-28T04:14:53.432Z        INFO    livekit service/turn.go:145     Starting TURN server    {"turn.relay_range_start": 30000, "turn.relay_range_end": 40000, "turn.portTLS": 5349, "turn.externalTLS": true, "turn.portUDP": 443}
2025-01-28T04:14:53.433Z        WARN    livekit service/server.go:146   prometheus_port is deprecated, please switch prometheus.port instead
2025-01-28T04:14:53.439Z        INFO    livekit service/server.go:258   starting LiveKit server {"portHttp": 7880, "nodeID": "ND_3vr7fHjkqWx3", "nodeIP": "212.46.61.44", "version": "1.8.3", "rtc.portTCP": 7881, "rtc.portICERange": [50000, 60000], "portPrometheus": 6789}



2025-01-28T04:14:35.231Z        WARN    livekit service/server.go:146   prometheus_port is deprecated, please switch prometheus.port instead
2025-01-28T04:14:35.237Z        INFO    livekit service/server.go:258   starting LiveKit server {"portHttp": 7880, "nodeID": "ND_qSMc6Re4U3dm", "nodeIP": "212.46.61.44", "version": "1.8.3", "rtc.portTCP": 7881, "rtc.portICERange": [50000, 60000], "portPrometheus": 6789}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x12c65b8]

goroutine 259 [running]:
github.com/livekit/livekit-server/pkg/service.(*defaultSessionHandler).HandleSession(0xc001054cc0, {0x1909038, 0xc0017fbc50}, {{0xc0005ae2b0, 0x9}, {0x0, 0x0}, 0x0, 0x0, 0x1, ...}, ...)
```

**Server**
- Version: 1.8.3
- Environment: Docker, deployed using docker-compose

livekit.yaml

```
port: 7880
log_level: warning
logging:
    pion_level: info
rtc:
    tcp_port: 7881
    port_range_start: 50000
    port_range_end: 60000
    use_external_ip: true
redis:
    address: 10.5.86.1:6379
turn:
    enabled: true
    domain: 4.lk-turn.xxxx.com
    cert_file: ""
    key_file: ""
    tls_port: 5349
    udp_port: 443
    external_tls: true
keys:
    XXXX: XXXXX
prometheus_port: 6789
```
I have reverted back to version 1.7.2 for now, and it works like a charm. The issue only exists with version 1.8.x. Any pointers on fixing this would be much appreciated , Thanks! 🙏🏻

--- a/pkg/routing/interfaces.go
+++ b/pkg/routing/interfaces.go
@@ -256,13 +256,6 @@ func ParticipantInitFromStartSession(ss *livekit.StartSession, region string) (*
 		return nil, err
 	}
 
-	// TODO: clean up after 1.7 eol
-	if ss.CreateRoom == nil {
-		ss.CreateRoom = &livekit.CreateRoomRequest{
-			Name: ss.RoomName,
-		}
-	}
-
 	pi := &ParticipantInit{
 		Identity:        livekit.ParticipantIdentity(ss.Identity),
 		Name:            livekit.ParticipantName(ss.Name),
@@ -280,6 +273,13 @@ func ParticipantInitFromStartSession(ss *livekit.StartSession, region string) (*
 	if ss.SubscriberAllowPause != nil {
 		subscriberAllowPause := *ss.SubscriberAllowPause
 		pi.SubscriberAllowPause = &subscriberAllowPause
+	}
+
+	// TODO: clean up after 1.7 eol
+	if pi.CreateRoom == nil {
+		pi.CreateRoom = &livekit.CreateRoomRequest{
+			Name: ss.RoomName,
+		}
 	}
 
 	return pi, nil

--- a/pkg/routing/interfaces.go
+++ b/pkg/routing/interfaces.go
@@ -256,6 +256,13 @@ func ParticipantInitFromStartSession(ss *livekit.StartSession, region string) (*
 		return nil, err
 	}
 
+	// TODO: clean up after 1.7 eol
+	if ss.CreateRoom == nil {
+		ss.CreateRoom = &livekit.CreateRoomRequest{
+			Name: ss.RoomName,
+		}
+	}
+
 	pi := &ParticipantInit{
 		Identity:        livekit.ParticipantIdentity(ss.Identity),
 		Name:            livekit.ParticipantName(ss.Name),

